python/lsst/daf/butler/_butler.py

```diff
@@ -371,7 +376,10 @@ def makeRepo(root: str, config: Union[Config, str, None] = None, standalone: boo
         config.dumpToUri(configURI, overwrite=overwrite)
 
         # Create Registry and populate tables
-        Registry.fromConfig(config, create=createRegistry, butlerRoot=root)
+        if createRegistry:
```

Comment:
I see three places where we use createRegistry=False and those are just in tests. Given what we are currently doing with registry versioning I don't think setting this to False is ever right. There is further discussion on DM-26237.

One of them seems to involve creating a registry, then overwriting the config file with the standalone config (but in the same registry) and then comparing configs. That test could run in two distinct butlers so doesn't need createRegistry=False. The other one in test_butler is solely for testing overwrite=False (so the createRegistry is irrelevant). Finally there is one usage in test_testRepo that uses makeRepo to create a butler.yaml outside of the repository and doesn't care about the registry itself being set up -- I think that could be True without any problem.

Reply:
I'll drop `createRegistry`, indeed it makes no sense to try to keep it any longer.